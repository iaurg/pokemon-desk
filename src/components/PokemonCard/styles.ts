import styled, { css } from 'styled-components'

type ColorObject = {
  [key: string]: string
}

const typesColors: ColorObject = {
  normal: '#A8A878',
  fire: '#F08030',
  water: '#6890F0',
  electric: '#F8D030',
  grass: '#78C850',
  ice: '#98D8D8',
  fighting: '#C03028',
  poison: '#A040A0',
  ground: '#E0C068',
  flying: '#A890F0',
  psychic: '#F85888',
  bug: '#A8B820',
  rock: '#B8A038',
  ghost: '#705898',
  dragon: '#7038F8',
  dark: '#705848',
  steel: '#B8B8D0',
  fairy: '#EE99AC'
}

export const Wrapper = styled.main`
  background-color: #fff;
  color: #000;
  padding: 1.5rem 2rem;
  text-align: left;
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.1);
  border-radius: 0.8rem;
  width: fit-content;
  display: flex;
  transition: all 0.2s ease-in-out;
  align-items: center;
  margin: 1rem;
  &:hover {
    box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.2);
    cursor: pointer;
    transition: all 0.2s ease-in-out;
  }
  width: 220px;
  justify-content: space-between;
`

export const Content = styled.div`
  margin-right: 0.5rem;
`

export const Name = styled.h1`
  color: #081837;
  padding-bottom: 1rem;
`

export const Types = styled.div``

export const Type = styled.span`
  ${({ color }) => css`
    background-color: ${typesColors[color ? color : 'normal']};
    color: #fff;
    text-transform: uppercase;
    padding: 0.5rem;
    margin-right: 0.5rem;
    border-radius: 0.5rem;
  `}
`

export const Image = styled.img`
  width: 6rem;
  height: 6rem;
`
