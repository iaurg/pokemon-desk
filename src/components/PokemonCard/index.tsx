import * as S from './styles'

export type PokemonType = {
  slot: number
  type: {
    name: string
  }
}

export type PokemonCardProps = {
  name: string
  types: PokemonType[]
  image: string
}

const PokemonCard = ({ name, types, image }: PokemonCardProps) => (
  <S.Wrapper>
    <S.Content>
      <S.Name>{name}</S.Name>
      <S.Types>
        {types.map(({ type: { name: typeName } }, index) => (
          <S.Type color={typeName} key={index}>
            {typeName}
          </S.Type>
        ))}
      </S.Types>
    </S.Content>
    <S.Image src={image} alt={name} />
  </S.Wrapper>
)

export default PokemonCard
