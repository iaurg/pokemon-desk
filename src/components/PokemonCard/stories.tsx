import { Story, Meta } from '@storybook/react'
import PokemonCard, { PokemonCardProps } from '.'

export default {
  component: PokemonCard,
  args: {
    name: 'bulbasaur',
    image:
      'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png',
    types: [
      {
        slot: 1,
        type: {
          name: 'grass'
        }
      },
      {
        slot: 2,
        type: {
          name: 'poison'
        }
      }
    ]
  }
} as Meta<PokemonCardProps>

export const Default: Story<PokemonCardProps> = (args) => (
  <PokemonCard {...args} />
)
