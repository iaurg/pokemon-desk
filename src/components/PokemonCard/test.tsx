import { render, screen } from '@testing-library/react'

import PokemonCard from '.'

const props = {
  name: 'bulbasaur',
  image:
    'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png',
  types: [
    {
      slot: 1,
      type: {
        name: 'grass'
      }
    },
    {
      slot: 2,
      type: {
        name: 'poison'
      }
    }
  ]
}

describe('<PokemonCard />', () => {
  it('should render the component', () => {
    const { container } = render(<PokemonCard {...props} />)

    expect(
      screen.getByRole('heading', { name: /bulbasaur/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('img', { name: /bulbasaur/i })).toBeInTheDocument()

    expect(screen.getByText(/grass/i)).toBeInTheDocument()
    expect(screen.getByText(/poison/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
