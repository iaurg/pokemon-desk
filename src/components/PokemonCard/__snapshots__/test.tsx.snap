// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonCard /> should render the component 1`] = `
.c0 {
  background-color: #fff;
  color: #000;
  padding: 1.5rem 3rem;
  text-align: left;
  box-shadow: 0 0.5rem 1rem rgba(0,0,0,0.1);
  border-radius: 0.8rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 1rem;
}

.c0:hover {
  box-shadow: 0 0.5rem 1rem rgba(0,0,0,0.2);
  cursor: pointer;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c1 {
  margin-right: 1rem;
}

.c2 {
  color: #081837;
  padding-bottom: 1rem;
}

.c3 {
  background-color: #78C850;
  color: #fff;
  text-transform: uppercase;
  padding: 0.5rem;
  margin-right: 0.5rem;
  border-radius: 0.5rem;
}

.c4 {
  background-color: #A040A0;
  color: #fff;
  text-transform: uppercase;
  padding: 0.5rem;
  margin-right: 0.5rem;
  border-radius: 0.5rem;
}

.c5 {
  max-width: 8rem;
  height: 8rem;
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <h1
      class="c2"
    >
      bulbasaur
    </h1>
    <div
      class=""
    >
      <span
        class="c3"
        color="grass"
      >
        grass
      </span>
      <span
        class="c4"
        color="poison"
      >
        poison
      </span>
    </div>
  </div>
  <img
    alt="bulbasaur"
    class="c5"
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png"
  />
</main>
`
